src/extension.ts

```diff
@@ -52,7 +52,12 @@ function setupEvents(rockContext: context.Context, extensionContext: vscode.Exte
                 handleNewWorkspaceFolder(folder.uri.fsPath, rockContext, workspaces, configManager);
             });
             event.removed.forEach((folder) => {
-                workspaces.deleteFolder(folder.uri.fsPath);
+                let deletedWs = workspaces.deleteFolder(folder.uri.fsPath);
+                if (deletedWs) {
+                    deletedWs.readWatchPID().
+                        then((pid) => process.kill(pid)).
```

Comment:
This defaults to SIGTERM.. Wouldn't it be better to send a SIGINT instead?

Reply:
Good catch, I'll update